app/backends/workflow-orchestration.ts

```diff
@@ -159,16 +159,18 @@ export async function updateWorkItem(req: HarmonyRequest, res: Response): Promis
     // If the response is an error then set the job status to 'failed'
     } else if (workItem.status === WorkItemStatus.FAILED) {
       if (![JobStatus.FAILED, JobStatus.CANCELED].includes(job.status)) {
-        job.status = JobStatus.FAILED;
         let message: string;
         if (errorMessage) {
           message = `WorkItem [${workItem.id}] failed with error: ${errorMessage}`;
         } else {
           message = 'Unknown error';
         }
-        job.message = message;
+        job.fail(message);
         try {
           await job.save(tx);
+          await updateWorkItemsStatusByJobId(
```

Comment:
I'm wondering if this should all be taken a step further and put in a method on `job`.  Something like `job.complete(tx, finalStatus, message)`.  For that matter, I was a little surprised that this all didn't end up inside the `fail` method and that we have to save and update stuff after calling `fail`.  Maybe that's similar to your question from the stand up about the behavior of `save`.

Either way, I think a method on `Job` that knows all the non-terminal statuses could be helpful if we ever end up adding one